### PR TITLE
WIP: Add KubeVirt rate limiting profile `kubelet`

### PR DESCRIPTION
**Pull request by jcanocan**
**What this PR does / why we need it**:
The rate limiting profile `kubelet` allows users to dynamically set the provided kubelet `KubeAPIQPS` and `KubeAPIBurst` to KubeVirt components:  `API`, `Webhook`, `Controller` and `Handler`. The profile could be enabled adding the following to the HCO definition:

```yaml
apiVersion: hco.kubevirt.io/v1beta1
kind: HyperConverged
...
spec:
  tuningPolicy: kubelet
...
```

This profile relies on the [Machine Configuration Operator](https://github.com/openshift/machine-config-operator) (MCO) to detect changes on the kubelet configuration. Therefore, MCO is required in order to work. This operator is available by default on OpenShift except for HyperShift where it's not available. Also, among all MCO objects, the `kubelet` profile uses the `XX-worker-generated-kubelet` [objects](https://github.com/openshift/machine-config-operator/pull/2547) to fetch and keep track of the rate limiting values.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-28519
```

**Release note**:

```release-note
Add `kubelet` rate limting profile
```

### CODE REVIEW

**Comment by jcanocan** [comment]
This is a WIP, the idea behind this PR is to get feedback from the community and to find out the best way to get this feature. Alternatives to MCO are listed in [CNV-28519](https://issues.redhat.com/browse/CNV-28519).

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type HyperConvergedTuningPolicy string
 const (
 	HyperConvergedAnnotationTuningPolicy HyperConvergedTuningPolicy = "annotation"
 	HyperConvergedHighBurstProfile       HyperConvergedTuningPolicy = "highBurst"
+	HyperConvergedKubeletProfile         HyperConvergedTuningPolicy = "kubelet"
```
Comment (fabiand) [inline api/v1beta1/hyperconverged_types.go:35]: What about `follow-kubelet`

Comment (jcanocan) [reply api/v1beta1/hyperconverged_types.go:35]: Sounds good to me! We have also considered naming it as something like `mcoKubelet`, with the idea of reflecting the dependency of the machine configuration operator (mco). Another possibility is a merge between both ideas like: `followMCOKubelet`. WDYT? In any case, `followKubelet` (IMHO without the `-` for consistency purposes) sounds better than just `kubelet` :)

**Pull request closed without merging**